OpenEdXMobile/src/main/java/org/edx/mobile/view/AccountActivity.java

```diff
@@ -2,26 +2,33 @@
 
 import android.content.Context;
 import android.content.Intent;
+import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 
 import org.edx.mobile.base.BaseSingleFragmentActivity;
+import org.edx.mobile.deeplink.ScreenDef;
 import org.edx.mobile.logger.Logger;
 import org.edx.mobile.util.Config;
 
 import javax.inject.Inject;
 
+import static org.edx.mobile.view.Router.EXTRA_SCREEN_NAME;
+
 public class AccountActivity extends BaseSingleFragmentActivity {
     protected Logger logger = new Logger(getClass().getSimpleName());
 
     @Inject
     private Config config;
 
-    public static Intent newIntent(Context activity) {
-        return new Intent(activity, AccountActivity.class);
+    public static Intent newIntent(Context activity, @Nullable @ScreenDef String screenName) {
+        final Intent intent = new Intent(activity, AccountActivity.class);
+        intent.putExtra(EXTRA_SCREEN_NAME, screenName);
+        intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
```

Comment:
Please describe some use case for `FLAG_ACTIVITY_REORDER_TO_FRONT`, the application is working fine without this.

Reply:
Only 1 instance of AccountActivity should be present in back-stack any time. It will make sure of this.